Convert capnproto-feedstock to v1 feedstock

munechika-koyo opened the pull request:
This PR converts capnproto-feedstock to a v1 recipe and switch the conda build tool to rattler-build.
It has been automatically generated with [feedrattler v0.3.14](https://github.com/hadim/feedrattler).

Changes:
- [x] 📝 Converted `meta.yaml` to `recipe.yaml`
- [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
- [x] 🔧 Updated `conda-forge.yml` to use `rattler-build` and `pixi` (optional)
- [x] 🔢 Bumped the build number
- [x] 🐍 Applied temporary fixes for `python_min` and `python_version`
- [x] 🔄 Rerender the feedstock with conda-smithy
- [x] Ensured the license file is being packaged.

munechika-koyo committed:
Convert to v1 feedstock and use pixi as conda install tool

munechika-koyo committed:
MNT: Re-rendered with conda-smithy 3.53.3 and conda-forge-pinning 2025.12.08.19.17.28

Other tools:
- conda-build 25.11.1
- rattler-build 0.54.0
- rattler-build-conda-compat 1.4.10

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/recipe.yaml```) and found it was in an excellent condition.

munechika-koyo committed:
Add automerge configuration to bot settings in conda-forge.yml

munechika-koyo committed:
Refactor requirements and test commands for improved readability in recipe.yaml

munechika-koyo committed:
Remove unnecessary CMake options from build.bat

They are automatically offered.

munechika-koyo committed:
Add patch for pkg-config file installation on Windows

munechika-koyo committed:
Fix cmake command in build.bat to include CMAKE_ARGS for better configuration

munechika-koyo committed:
Reorder build requirements for clarity between Windows and Unix systems

munechika-koyo committed:
Add munechika-koyo to recipe maintainers list

munechika-koyo committed:
Update recipe and configuration files for capnproto-feedstock

- Add @munechika-koyo to CODEOWNERS and maintainers list
- Update version to 3.54.1 in pixi.toml
- Change context version to 1.3.0 in recipe.yaml
- Update source hash to sha256 in recipe.yaml
- Adjust build number and requirements in recipe.yaml
- Add checks for OSX_SDK_DIR in run_osx_build.sh

munechika-koyo committed:
Remove obsolete patch file pr-2456 from recipe

munechika-koyo commented:
This PR includes the v1.3.0 update.

munechika-koyo committed:
Fix indentation in recipe.yaml for test dependencies

munechika-koyo commented:
Hello,

I tried to convert the v0 recipe to v1 and reorganize its content, along with a bit of the build script.
However, I didn't find out why the osx-64 build failed, even though I shouldn't have changed the build script for that platform.

I would like to mark this PR as review stage and ask you to review it as well.
Thank you in advance.

jdblischak commented on conda-forge.yml:
```diff
@@ -12,3 +12,7 @@ provider:
+conda_install_tool: pixi
+conda_build_tool: rattler-build
+bot:
+  automerge: true
```
Please remove the bot automerge feature. This prevents us from having time to review the recipe and potentially updating the dependencies.

jdblischak reviewed (commented):
I agree it is puzzling why only the osx-64 build is failing. I'm not aware of any reason why rattler-build should differ from conda-build only on osx-64.

What is the motivation for this PR? Does the v1 recipe format or rattler-build provide some advantage for building capnproto?

munechika-koyo committed:
Remove bot configuration for automerge from conda-forge.yml

munechika-koyo committed:
Update build number to 1 in recipe.yaml

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I was trying to look for recipes to lint for you, but it appears we have a merge conflict. Please try to merge or rebase with the base branch to resolve this conflict.

Please ping the 'conda-forge/core' team (using the `@` notation in a comment) if you believe this is a bug.

munechika-koyo commented:
> What is the motivation for this PR? Does the v1 recipe format or rattler-build provide some advantage for building capnproto?

Thank you for your question!
I believe the v1 recipe format is more useful, especially because the `[tests.package_contents]` table makes it straightforward to understand and standardize it for any platform (currently `lib` section has an `if/then` statement since the windows build doesn't generate shared libraries, though.).
Also, rattler-build's build is faster than conda-build's, and it has a clearer log.

jdblischak commented:
> I believe the v1 recipe format is more useful, especially because the `[tests.package_contents]` table makes it straightforward to understand and standardize it for any platform (currently `lib` section has an `if/then` statement since the windows build doesn't generate shared libraries, though.).
> Also, rattler-build's build is faster than conda-build's, and it has a clearer log.

I agree those are good reasons in general to favor the v1 recipe format and rattler-build. And I appreciate you working on this!

That being said, I don't have a lot of bandwidth to maintain this feedstock, so I'm more in the camp of "If it ain't broke, don't fix it". If you or one of the other maintainers can fix the failing osx-64 build, I'm open to merging this PR. But after a quick skim of your changes, I don't see anything obvious to tweak to troubleshoot this further.

jdblischak commented:
> I was trying to look for recipes to lint for you, but it appears we have a merge conflict. Please try to merge or rebase with the base branch to resolve this conflict.

@munechika-koyo please rebase your PR onto main to incorporate #63 and then force push , which should fix the CI builds

munechika-koyo committed:
Merge branch 'main' into convert_feedstock_to_v1_recipe_format

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/recipe.yaml```) and found it was in an excellent condition.

munechika-koyo committed:
Set binutils requirement for only Linux

munechika-koyo commented:
@jdblischak Finally, all CIs passed successfully.
According to AI suggestions, `binutils` appeared to have misbehaved on osx-64, so I moved it into only linux dependencies.

jdblischak commented:
@conda-forge-admin, please restart ci

jdblischak commented on recipe/build.bat:
```diff
@@ -1,12 +1,9 @@
-cmake ^
+cmake %CMAKE_ARGS% ^
     -G "NMake Makefiles" ^
-    -DCMAKE_BUILD_TYPE=Release ^
```
I confirmed this is the default in the conda-forge env var `CMAKE_ARGS`

jdblischak commented on recipe/build.bat:
```diff
@@ -1,12 +1,9 @@
     -G "NMake Makefiles" ^
-    -DCMAKE_BUILD_TYPE=Release ^
     -DCMAKE_POSITION_INDEPENDENT_CODE=1 ^
-    -DCMAKE_INSTALL_PREFIX="%LIBRARY_PREFIX%" ^
```
I confirmed this is the default in the conda-forge env var `CMAKE_ARGS`

jdblischak commented on recipe/build.bat:
```diff
@@ -1,12 +1,9 @@
-    -DCMAKE_BUILD_TYPE=Release ^
     -DCMAKE_POSITION_INDEPENDENT_CODE=1 ^
-    -DCMAKE_INSTALL_PREFIX="%LIBRARY_PREFIX%" ^
-    -DCMAKE_INSTALL_LIBDIR=lib ^
```
I confirmed that the default for [`CMAKE_INSTALL_LIBDIR`](https://cmake.org/cmake/help/latest/module/GNUInstallDirs.html) is either `lib` or `lib64`. Given that we test in the recipe that it uses `lib`, this is clearly also the default used

jdblischak commented on recipe/recipe.yaml:
```diff
@@ -0,0 +1,103 @@
+        - libtool
+    - if: linux
+      then:
+        - binutils
```
`binutils` is a build-only dependency, and the non-linux builds succeed without it, so presumably it isn't needed

jdblischak reviewed (approved):
Thanks @munechika-koyo! Once the CI passes again, I'll merge this
